content/blog/scheduling-in-react/index.md

```diff
@@ -0,0 +1,243 @@
+---
+title: "Scheduling in React"
+date: "2019-03-15T12:00:00.000Z"
+---
+
+In modern applications, user interfaces often have to juggle multiple tasks at the same time. For example, a search component needs to respond to user input while providing auto completion results; An interactive dashboard needs to update the charts while loading data from the server and sending analytics data to your backend.
+
+All these parallel steps can lead to slow and unresponsive interfaces and unhappy users, so let’s learn how we can fix this.
+
+## Scheduling in User Interfaces
+
+Our users expect immediate feedback. Whether they are clicking on a button to open a modal or adding text to an input field, they don’t want to wait before seeing some kind of confirmation. For example, the button could show a modal and the input field could display the key that was typed.
+
+To visualize what happens when this is not the case, let’s take a look at the demo application that Dan Abramov presented at his talk, [Beyond React 16](https://reactjs.org/blog/2018/03/01/sneak-peek-beyond-react-16.html), at JSConf Iceland 2018.
+
+The application works like this: The more you type into the input below, the more detailed the charts below will get. Since both of the updates (the input element and the chart) run at the same time, the browser has to do so much computation that it will drop some frames. This leads to noticeable delays and a bad user experience:
+
+<video src="/blog/scheduling-in-react/sync-mode.mp4" muted="true" autoplay muted playsinline loop></video>
+
+However, a version that prioritizes updating the input with new keystrokes will appear to run a lot faster to the end user. This is because they receive immediate feedback even though the same computation time is required:
```

Comment:
Who is "they?" The keystrokes? The end user?

"This is because they receive immediate feedback even though the same computation time is required:"

Reply:
The users. That sentence is indeed unclear. Thanks!